snowshu/core/graph.py

```diff
@@ -18,8 +18,8 @@ def __init__(self):
         self.dag: tuple = None
         self.graph: networkx.Graph = None
 
-    def build_graph(self, configs: Configuration,
-                    full_catalog: Tuple[Relation]) -> networkx.DiGraph:
+    # TODO remove extra filtering of relations when building graph
```

Comment:
I'm not sure we ever needed this tbh? seems like an unreachable condition

Reply:
nvm I was looking for the broken dag filtering. yeah this can all be vastly simplified now methinks

Reply:
This comment is for the lines like [this one](https://github.com/eculicny/snowshu/blob/pre-filter-relations/snowshu/core/graph.py#L41), but there are more in the `_apply_specifications` function as well.

The broken dag filtering are checks like [this](https://github.com/eculicny/snowshu/blob/pre-filter-relations/snowshu/core/graph.py#L52;L45), unless I'm missing something. If you start with a digraph, it will always be directed. I assume the check should be something like `is_acyclic()` instead since the error message indicates cycles exist.